custom_components/tuya_local/devices/make_sky_blue_MPPT.yaml

```diff
@@ -0,0 +1,264 @@
+    dps:
+      - id: 108
+        type: integer
+        name: electric_total
+        unit: W
+        class: measurement
+        range:
+          min: 0
+          max: 100000
+        mapping:
+          - scale: 10
+        step: 1
+  - entity: number
+    name: uv_reco_value
+    category: config
+    dps:
+      - id: 109
+        type: integer
+        name: uv_reco_value
+        unit: V
+        mapping:
+          - scale: 10
+        step: 1
+        range:
+          min: 0
+          max: 1000
+  - entity: number
+    name: undervol_value
+    category: config
+    dps:
+      - id: 110
+        type: integer
+        name: undervol_value
+        unit: V
+        mapping:
+          - scale: 10
+        step: 1
+        range:
+          min: 0
+          max: 1000
+  - entity: sensor
+    category: diagnostic
+    class: battery
+    dps:
+      - id: 111
+        type: integer
+        name: battery_percentage
+        unit: "%"
+        ange:
+          min: 0
+          max: 100
+        mapping:
+          - scale: 10
+        step: 1
+  - entity: option
+    name: battery_type
+    category: config
+    dps:
+      - id: 112
+        class: enum
+        name: battery_type
+        mapping:
+          - dps_val: lead_acid
+            value: lead_acid
+          - dps_val: lithium
+            value: lithium
+  - entity: sensor
+    category: diagnostic
+    class: temperature
+    dps:
+      - id: 113
+        type: integer
+        name: temp_current
+        unit: ℃
+        range:
+          min: -400
+          max: 1200
+        mapping:
+          - scale: 10
+        step: 1
+  - entity: sensor
+    name: battery_cell_count
+    category: diagnostic
+    class: enum
+    dps:
+      - id: 114
+        type: integer
+        name: battery_cell_count
+        range:
+          min: 1
+          max: 20
+        scale: 0
+        step: 1
+  - entity: number
+    name: equalization_volt
+    category: config
+    class: voltage
+    dps:
+      - id: 115
+        type: integer
+        name: equalization_volt
+        unit: V
+        mapping:
+          - scale: 10
+        step: 1
+        range:
+          min: 0
+          max: 1000
+  - entity: number
+    name: equalizatifloat_charg_volton_volt
+    category: config
+    class: voltage
+    dps:
+      - id: 116
+        type: integer
+        name: float_charg_volt
+        unit: V
+        mapping:
+          - scale: 10
+        step: 1
+        range:
+          min: 0
+          max: 1000
+  - entity: number
+    name: constant_cur_set
+    category: config
+    class: current
+    dps:
+      - id: 117
+        type: integer
+        name: constant_cur_set
+        unit: A
+        scale: 0
+        step: 1
+        range:
+          min: 0
+          max: 150
+  - entity: number
+    name: output_timr
+    category: config
+    class: duration
+    dps:
+      - id: 1189
+        type: integer
+        name: output_timr
+        unit: h
+        scale: 0
```

Comment:
Never use a scale of 0, Tuya uses it to mean no scaling, but here that would be 1 and is better left out. A scale of 0 will result in divide by 0 error.